rpc/ws/client.go

```diff
@@ -98,6 +93,31 @@ func ConnectWithOptions(ctx context.Context, rpcEndpoint string, opt *Options) (
 	return c, nil
```

Comment:
If both retry backoff in  `c.receiveMessages()` returns nothing when we lost network connection for 30 minutes (for example), then we should handle this case. 
Because at that time, we have meaningless goroutine which come from line 82-88 though we have receiveMessages stopped. And our scanners will not receive new data from ws when the server has internet.

Imo, we should pick our strategy for this problem, whether reconnect ws after time then panic or reconnect forever. Then re-write logic flow according to it.

Reply:
This is a lib so we should not panic here. I'll send an error to `Subscription.err` to notify the backend.